dl1_data_handler/reader.py

```diff
@@ -1697,6 +1774,564 @@ def _append_features(self, batch) -> Table:
+            # Compute rolling sum using cumsum
+            start = np.arange(time) - tdscan_settings["eps_t"]
+            end = np.arange(time) + tdscan_settings["eps_t"] + 1
+
+            start = np.clip(start, 0, time)
+            end = np.clip(end, 0, time)
+
+            # Get total values in the window for each time frame using broadcasting
+            total = cumsum_l1[neighbors][:, end] - cumsum_l1[neighbors][:, start]  # shape: (num_neighbors, time)
+            # sum across neighbors and threshold.
+            tdscan_flowers[i] = (np.sum(total, axis=0) >= tdscan_settings["min_pts"]).astype(np.float32)
+        bin_flowers = tdscan_flowers # shape: (n_flowers, time)
+
+    return bin_flowers
+
+
+class DLRawTriggerReader(DLWaveformReader):
+    """
+    A data reader class for handling R0 raw waveform data from telescopes.
+
+    This class extends the ``DLWaveformReader`` to specifically handle the reading,
+    transformation, and mapping of R0 raw waveform data. It supports ``mono`` data loading 
+    mode and can multiple output settings based on square trigger patches.
+
+    Attributes
+    ----------
+    output_settings : str
+        Specifies the output format in the batch. Can be ``waveform``, ``balanced_patches`` or ``all_patches``.
+    trigger_settings : dict
+        Contains settings for trigger, including the threshold in photoelectrons, the number of thrigger patches,
+        the position of the center of the trigger patches. 
+    """
+
+    output_settings = CaselessStrEnum(
+        ["waveform", "balanced_patches", "all_patches", "double_patches", "max_patches"],
+        default_value="waveform",
+        help=(
+            "Set the way to retrieve data from each event. "
+            "``waveform``: extract the sequence of selected samples for the complete camera. "
+            "``balanced_patches``: extract the sequence of selected samples for equilibrated number of cosmic and nsb patches. "
+            "``all_patches``: extract the sequence of selected samples for all patches. "
+            "``double_patches``: extract the sequence of selected samples for a random nsb patch and the patch containing the brightest pixel. "
+            "``max_patches``: extract all the nsb patches and all the patches above the cpe_threshold. "
+        ),
+    ).tag(config=True)
+
+    output_size = CaselessStrEnum(
+        ["patch", "sector", "camera"],
+        default_value = "camera",
+        help=(
+            "Set the number of pixels in the output vector, patch and sector only available for the Advanced SiPM camera. "
+            "``patch``: 343 pixels trigger patches."
+            "``sector``: 29898 pixels patches, approx. 1/3 of the camera. "
+            "``camera``: complete camera. "
+        ),
+    ).tag(config=True) 
+
+    add_dummy_channel = Bool(
+        default_value = False,
+        allow_none = False,
+        help=("Boolean variable to add or not an extra dummy dimension for the channel")
```

Comment:
Please add a little bit more information for this help. E.g. why this is needed/when the user needs to do this action oon swiitching the boolean on.